R/path.R

```diff
@@ -0,0 +1,273 @@
+#' 
+#' * `tables_to()` sets the `pmtables.dir` option as the default table output
+#'   directory; this function also sets the `pmtables.path.type` option, 
+#'   defaulting to `proj`.
+#'   
+#' * `figures_to()` sets the `mrggsave.dir` option as the default figure output 
+#'   directory.
+#'   
+#' * `tf_options()` prints the `mrg.script`, `pmtables.dir`, and `mrggsave.dir`
+#'   options to the console.
+#'   
+#' * `tf_options_clear()` clears options set by `mrg_script()`, `tables_to()`, 
+#'   and `figures_to()`, setting the option value to `NULL`.
+#'   
+#' @param path the absolute path or path relative to the current working
+#' directory.
+#' @param path.type indicates how `pmtables` should format the path portion
+#' of table annotations; see `pmtables::format_table_path`. 
+#' @param set_script if `TRUE` (the default), include a call to `mrg_script()` 
+#' to set the `mrg.script` option.  
+#' @param quietly if `TRUE`, don't echo `tf_options()` outputs after options
+#' are cleared with `tf_options_clear()`.
+#'  
+#' @return
+#' - `tables_to()`, `figures_to()` and `mrg_script()` return the value of the 
+#'   option invisibly
+#' - `tf_options()` prints messages to the console and returns `NULL` invisibly
+#' - `tf_options_clear()` prints `tf_options()` output unless the user requests
+#'   quiet reset.
+#'  
+#' @md
+#' @rdname tf_options
+#' @export
+tf_options <- function() {
+  check_path_deps()
+
+  not_set <- "<option not set>"
+  missing <- "<does not exist>"
+  
+  proj <- this_proj()
+
+  # mrg.script
+  script <- options()$mrg.script
+  if(!is.character(script)) {
+    script <- not_set 
+  } else if (!file.exists(file.path(proj, script))) {
+    script <- paste(script, missing)
+  }
+  script <- paste0("script  ", script)
+  names(script) <- "*"
+  inform(script)
+  
+  # pmtables.dir
+  tables <- options()$pmtables.dir
+  if (!is.character(tables)) {
+    tables <- not_set
+  } else if (dir.exists(tables)) {
+    tables <- proj_rel(tables)
+  } else {
+    tables <- paste(tables, missing)
+  }
+  tables <- paste0("tables  ", tables)
+  names(tables) <- "*"
+  inform(tables)
+  
+  # mrggsave.dir
+  figures <- options()$mrggsave.dir
+  if (!is.character(figures)) {
+    figures <- not_set
+  } else if (dir.exists(figures)) {
+    figures <- proj_rel(figures)
+  } else {
+    figures <- paste(figures, missing)
+  }
+  figures <- paste0("figures ", figures)
+  names(figures) <- "*"
+  inform(figures)
+  
+  invisible(NULL)
+}
+
+#' @rdname tf_options
+#' @export
+tf_options_clear <- function(quietly = FALSE) {
+  check_path_deps()
+
+  options(
+    mrg.script = NULL, 
+    pmtables.dir = NULL, 
+    pmtables.path.type = NULL,
+    mrggsave.dir = NULL
+  )
+  if(!isTRUE(quietly)) {
+    tf_options()
+  }
+  return(invisible(NULL))
+}
+
+#' @rdname tf_options
+#' @export
+mrg_script <- function(path = NULL) {
+  check_path_deps()
+
+  if(is.null(path)) {
+    path <- this_file_proj()  
```

Comment:
Is `mrg.script` typically relative to the project rather than the base name of the script?  In the FOCE example project, it looks like it's always the base name.

Reply:
@kyleam it was historically just the base name b/c we used to keep every script in a single directory. Now stuff is all over the place, so one of the primary goals of all of this is to get the path from project root all the way to the script. This should _only_ show up in table and figure annotations as far as I know at this time.